lib/gst/player/gstplayer.c

```diff
@@ -1121,6 +1149,594 @@ element_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
+{
+  GstPlayerVideoInfo *info = (GstPlayerVideoInfo*) stream_info;
+
+  if (stream_info->caps) {
+    GstStructure *s;
+
+    s = gst_caps_get_structure (stream_info->caps, 0);
+    if (s) {
+      gint width, height;
+      gint fps_n, fps_d;
+
+      if (gst_structure_get_int (s, "width", &width))
+        info->width = width;
+
+      if (gst_structure_get_int (s, "height", &height))
+        info->height = height;
+
+      if (gst_structure_get_fraction (s, "framerate", &fps_n, &fps_d)) {
+        info->framerate_num = fps_n;
+        info->framerate_denom = fps_d;
+      }
+    }
+  }
+
+  if (stream_info->tags) {
+    guint bitrate, max_bitrate;
+
+    if (gst_tag_list_get_uint (stream_info->tags, GST_TAG_BITRATE, &bitrate) ||
+        gst_tag_list_get_uint (stream_info->tags, GST_TAG_NOMINAL_BITRATE,
+                                &bitrate))
+      info->bitrate = bitrate;
+
+    if (gst_tag_list_get_uint (stream_info->tags, GST_TAG_MAXIMUM_BITRATE,
+                                &max_bitrate))
+      info->max_bitrate = max_bitrate;
+  }
+}
+
+static void
+update_audio_info (GstPlayer *self, GstPlayerStreamInfo *stream_info)
+{
+  GstPlayerAudioInfo *info = (GstPlayerAudioInfo*) stream_info;
+
+  if (stream_info->caps) {
+    GstStructure *s;
+
+    s = gst_caps_get_structure (stream_info->caps, 0);
+    if (s) {
+      gint rate, channels;
+
+      if (gst_structure_get_int (s, "rate", &rate))
+        info->sample_rate = rate;
+
+      if (gst_structure_get_int (s, "channels", &channels))
+        info->channels = channels;
+    }
+  }
+
+  if (stream_info->tags) {
+    guint bitrate, max_bitrate;
+    gchar *lang_code = NULL;
+
+    if (gst_tag_list_get_uint (stream_info->tags, GST_TAG_BITRATE, &bitrate) ||
+        gst_tag_list_get_uint (stream_info->tags, GST_TAG_NOMINAL_BITRATE,
+                                &bitrate))
+      info->bitrate = bitrate;
+
+    if (gst_tag_list_get_uint (stream_info->tags, GST_TAG_MAXIMUM_BITRATE,
+                                &max_bitrate))
+      info->max_bitrate = max_bitrate;
+
+    if (gst_tag_list_get_string (stream_info->tags, GST_TAG_LANGUAGE_CODE,
+          &lang_code)) {
+      info->language = g_strdup (gst_tag_get_language_name (lang_code));
+      g_free (lang_code);
+    }
+    else {
+      gst_tag_list_get_string (stream_info->tags, GST_TAG_LANGUAGE_CODE,
+          &info->language);
+    }
+  }
+}
+
+static GstPlayerStreamInfo *
+update_stream_info (GstPlayer *self, gint stream_id, GType type,
+  GstTagList *tags)
+{
+  GstTagList  *result = NULL;
+  GstPlayerStreamInfo  *stream_info;
+
+  stream_info = get_stream_info (self, type, stream_id);
+
+  if (stream_info == NULL) {
+    /* create a new stream object */
+    stream_info = stream_info_new (self, type, stream_id);
+
+    if (stream_info == NULL) {
+      return NULL;
+    }
+  }
+
+  result = gst_tag_list_merge (stream_info->tags, tags,
+                               GST_TAG_MERGE_REPLACE);
```

Comment:
I think the taglist needs to be completely replaced. What was stream X before, might now be a completely different stream and have nothing to do at all anymore with the previous one